block/s3/inventory_iterator.go

```diff
@@ -45,12 +45,16 @@ func NewInventoryIterator(ctx context.Context, inv *Inventory, invBucket string)
 	}
 	res.rowsPerFile = make([]int, len(res.Manifest.Files))
 	for i := range res.Manifest.Files {
-		pr, closeReader, err := res.getParquetReader(res.ctx, res.S3, res.inventoryBucket, res.Manifest.Files[i].Key)
+		key := res.Manifest.Files[i].Key
+		pr, closeReader, err := res.getParquetReader(res.ctx, res.S3, res.inventoryBucket, key)
 		if err != nil {
 			return nil, err
 		}
 		res.rowsPerFile[i] = int(pr.GetNumRows())
-		_ = closeReader()
+		err = closeReader()
+		if err != nil {
+			res.logger.Errorf("failed to close parquet reader in NewInventoryIterator. bucket=%s, key=%s", invBucket, key)
```

Comment:
Structured logging is often more observable:
```suggestion
			res.logger.WithFields(logging.Fields{"bucket":  invBucket, "key": key}).
			    Error("failed to close parquet reader in NewInventoryIterator)
```
Now queries are easy, and even statistics etc. with ES and friends.

Also below.

Reply:
nice! thanks.